### DISCUSSION TITLE
Use GUN for p2p syncing

### DISCUSSION THREAD

**Discussion by davay42** [davidmyersdev/octo#111] [Ideas]
Have you seen [GUNJS](https://gun.eco) already? It's a reactive p2p graph database, that's working blazing fast. It may become the wire, connecting notes between user's different devices and also making them shareable with other users.

**Comment by davidmyersdev** [comment]
I did a little bit of reading into this. This seems like a really interesting idea, but I think it might make more sense as an extension to octo that users can opt-in to. Once the octo API stabilizes, I hope to build out an extension system that anyone can contribute to. I'm going to leave this open for now, because I'd love to continue the discussion and hear your thoughts!

**Comment by davay42** [comment]
Sounds reasonable. Gun is in active development right now and should become more stable in some near future. For all of my research on actual decentralized p2p web systems I find it the most easy, fast and straightforward. It uses native browser crypto functions to provide a simple yet powerful SEA.js library. I have experimented with it a little and it's a very powerful tool.
I admire your project with such a nice and clean Vue structure. That's what I'm trying to come up with in all my recent projects, but always got buried under growing complexity of the app. I was hoping that Vue 3 Composition API may help make my structures more easy and reliable, but now it feels even more complicated. ))

**Comment by davidmyersdev** [comment]
I really appreciate that! 😄 

It has definitely taken some time to get it to this point (with many iterations). Things were much messier before I open sourced this project. I've actually started looking into [Nuxt.js](https://nuxtjs.org/), because it offers a more comprehensive structure when using all the Vue libraries together (Vuex & VueRouter). It will depend on the level of effort, but I would really like to migrate to Nuxt.js when I upgrade octo to Vue 3. You can follow that progress at #44.
